bootstrap/14-boot-partition.sh

```diff
@@ -20,31 +20,47 @@
 
 check_if_variable_is_set SOURCE_DIR
 
-# Get the files, which must present on the boot partition, from different
-# sources and put them in the directory specified via BOOT.
-boot="`get_attr ${OS} boot`"
-for f in ${boot}; do
-    if [ ! -z `echo ${f} | egrep "^https://|^http://|^ftp://"` ]; then
-        info "downloading ${f} to ${BOOT}"
-        wget -q -O ${BOOT}/`basename ${f}` ${f}
-    elif [[ ${f:0:1} == "/" ]]; then
-        # Split the name of the target file or directory into two parts:
-        # original name and copy name.
-        IFS=':' read -ra FILE_NAMES <<< "${R}/${f}"
-
-        info "copying ${FILE_NAMES[0]} to ${BOOT}"
-
-        if [ -z ${FILE_NAMES[1]} ]; then
-            # If the name of the copy is not specified, use the original one.
-            cp -r ${FILE_NAMES[0]} ${BOOT}
+if [ ! -z ${BOOT_DIR} ] && [ -d ${BOOT_DIR} ]; then
+    info "using ${BOOT_DIR} as a source for the boot partition."
+    cp -r --preserve "${BOOT_DIR}"/. ${BOOT}
```

Comment:
I think it's better to use `rsync -a "${BOOT_DIR}"/ ${BOOT}`, instead.
The point is that in one similar case we use `rsync`, but in this case for some reason we decide to use `cp`. It means there is no generality.